python/rmm/rmm/_cuda/CMakeLists.txt

```
@@ -17,3 +17,10 @@ set(linked_libraries rmm::rmm)
 
 rapids_cython_create_modules(SOURCE_FILES "${cython_sources}" LINKED_LIBRARIES "${linked_libraries}"
                                                                                CXX)
+
+# mark all symbols in these Cython targets "hidden" by default, so they won't collide with symbols
+# loaded from other DSOs
+foreach(_cython_target IN LISTS RAPIDS_CYTHON_CREATED_TARGETS)
+  set_target_properties(${_cython_target} PROPERTIES C_VISIBILITY_PRESET hidden
+                                                     CXX_VISIBILITY_PRESET hidden)
```

All the targets are C++ so setting C visibility is superfluous (not harmful though).
```suggestion
  set_target_properties(${_cython_target} PROPERTIES CXX_VISIBILITY_PRESET hidden)
```

---

> (not harmful though)

This is what I was thinking. It's not harmful to have it here, and in exchange we don't need to think carefully here about whether a C compiler is ever invoked for any of these targets.

I think `C_VISIBILITY_PRESET` should be kept.

---

> I think `C_VISIBILITY_PRESET` should be kept.

I second this.

---

Sure that's OK with me.